### ISSUE TITLE
alsabat always return -2 when locale is C.UTF-8

### ISSUE THREAD

**Issue by miRoox** [alsa-project/alsa-utils#192]
alsabat always return -2 when locale is `C.UTF-8`, for example:
```sh
$ export LC_ALL=C.UTF-8
$ alsabat
alsa-utils version 1.2.6


Return value is -2
```

Note: Looks like `C.UTF-8` is the default locale if the system failed to set locale. For example, if I ssh from a device with `zh_CN.UTF-8` locale to a device that doesn't support this locale, the locale on the ssh session is `C.UTF-8`.

**Comment by marc-hb** [comment]
Thanks @miRoox , good catch.

This reproduces on Ubuntu 22 only when `C.utf8` is NOT in the locale archive. `C.utf8` is not in the archive by default.

`localedef --list` shows the locales currently built in the archive and `sudo dpkg-reconfigures locales` lets the user choose locales and rebuild the archive.

After selecting `C.UTF-8` in `sudo dpkg-reconfigures locales` and rebuilding, this `alsabat` failure goes away.

But wait: `C.utf8` seems very, very special and super confusing because there seems to be some very strange `C.utf-8` "fallback" hidden somewhere. Look at what happens when `C.utf8` is NOT in `localedef --list`:

- It is listed in `locale -a` anyway!
- `validlocale c.utf8` works anyway!
- `LANG=C.utf8 locale` works anyway!
- `LANG=C.utf-8 alsabat` works anyway even when...
- ... `LANG=C.utf8 alsabat` does not work!

Incredibly messy.

**Comment by marc-hb** [comment]
Note only `alsabat` fails, `aplay` is not affected. `alsabat -h` is not affected either.

`alsamixer` is very interesting. Both `C.utf8` and `C.utf-8` produce the same `alsamixer` output which is slightly better looking than `C.utf42` on my system.

So there is really something specific to `alsabat`.

Go figure
```diff
diff -u <(localedef --list ) <(locale -a )
--- /dev/fd/63	2023-02-22 19:19:48.186847345 -0800
+++ /dev/fd/62	2023-02-22 19:19:48.186847345 -0800
@@ -1,3 +1,5 @@
+C
+C.utf8
 en_AG
 en_AG.utf8
 en_AU.utf8
@@ -21,3 +23,4 @@
 en_ZM
 en_ZM.utf8
 en_ZW.utf8
+POSIX
```

**Comment by perexg** [comment]
Fixed in 55bafb79383dd2247a30d8e30e243e651f7b287f (@marc-hb : it was a clear bug). The problematic was setting `err = -errno` after `strdup()` which does not set errno (thus the error from the invalid locale settings was used).
